pkg/query/config.go

```diff
@@ -39,3 +44,29 @@ func LoadConfigs(confYAML []byte) ([]Config, error) {
 	}
 	return queryCfg, nil
 }
+
+// BuildQueryConfig returns a query client configuration from a static address.
+func BuildQueryConfig(queryAddrs []string) ([]Config, error) {
+	configs := make([]Config, 0, len(queryAddrs))
+	for _, addr := range queryAddrs {
+		if addr == "" {
+			return nil, errors.New("static querier address cannot be empty")
+		}
+		// If addr is missing schema, add http.
+		if !strings.Contains(addr, "://") {
```

Comment:
Rather than checking the presence of `://`, maybe parse the URL a first time here and check whether `u.Scheme` is empty or not?

Reply:
The problem is that if you omit scheme, it will parse your domian as scheme:
https://play.golang.org/p/doagHkCbLZu
Or can you suggestion how it can be done in other way @simonpasquier? From what I have checked on Go issues parse works like that for "invalid url" - the ones with missing sheme.

Reply:
You're right, sorry for the misleading comment.